### PR TITLE
Add empty iterator.

### PR DESCRIPTION
**Pull request by mcmtroffaes**
Taken from http://stackoverflow.com/a/13243870/2863746

### CODE REVIEW

File: more_itertools/more.py [hunk 1]
```diff
@@ -333,3 +333,12 @@ def intersperse(e, iterable):
...
             yield item
     raise StopIteration
+
+
+# http://stackoverflow.com/a/13243870/2863746
+def empty_iterator():
+    """
+    An empty iterator.
+    """
+    return
```
Comment (harlowja) [inline more_itertools/more.py]: Out of curosity why not just:
```
>>> def empty_iterator():
...   return iter(())
... 
```

Comment (mcmtroffaes) [inline more_itertools/more.py]: I like that solution better too (interestingly this was suggested on stackoverflow on the same day that I posted this patch!).

Comment (cvrebert) [inline more_itertools/more.py]: But that costs an extra function call vs. the current version

Comment (mcmtroffaes) [inline more_itertools/more.py]:
```
$ python -m timeit -s "def e(): return; yield" "e()"
1000000 loops, best of 3: 0.551 usec per loop
$ python -m timeit -s "def e(): return iter(())" "e()"
1000000 loops, best of 3: 0.252 usec per loop
$ python -m timeit "iter(())"
10000000 loops, best of 3: 0.148 usec per loop
```

**Comment by bbayles** [comment]
I'm inclined to close this in favor of the built-in `iter()`. Let me know if I'm missing something!

**Pull request closed without merging**